packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -350,6 +352,16 @@ class IosProject extends XcodeBasedProject {
         }
       }
     }
+
+    //When we add a watchOS target from Xcode 13, there is no Info.plist file inside the watchOS target folder.
+    //WkCompanionAppBundleIdentifier has been moved to project.pbxproj
+    final String? wkCompanionAppBundleIdentifier = firstMatchInFile(xcodeProjectInfoFile, _watchAppCompanionAppIdPattern)?.group(2);
```

Comment:
This isn't a good way to look up a build setting, it doesn't take into account different build configurations, multiple companion targets, variable substitution, etc. The _only_ time we do this is when you aren't on a Mac and we are last-ditch trying to figure out the bundle identifier, which isn't relevant in this case:
https://github.com/flutter/flutter/blob/42eb9032009b12a7671451b99a18828f3815b741/packages/flutter_tools/lib/src/xcode_project.dart#L204-L209

Instead, you should pipe through `buildSettings` from the caller and look up that way.

https://github.com/flutter/flutter/blob/8474f41e90c67dfffcbfbc6ba0edc3692b8d1846/packages/flutter_tools/lib/src/ios/mac.dart#L195

Reply:
@jmagman  Map returned from`buildSettingsForBuildInfo` doesn't contain all the build settings/configs in the Xcode project file. So I will not be able to get it from there. 

Did you mean to get the current build configuration(such as release/profile/debug) and then check the `INFOPLIST_KEY_WKCompanionAppBundleIdentifier` for that configuration rather than checking the first occurrence of `INFOPLIST_KEY_WKCompanionAppBundleIdentifier`. As of now, there is 3 occurrence of `INFOPLIST_KEY_WKCompanionAppBundleIdentifier` for Release, Profile & Debug respectively (When you add watchOS companion App, change is reflected in all build configurations).

Regarding tests, I will look into that. Not a dart expert here :)

Reply:
I have multiple custom build configs defined, and still am unable to build with these changes

Reply:
@kismet- once you apply this change to the xcode_project.dart file, have you deleted the /flutter/bin/cache/flutter_tools.snapshot to re-build the flutter tool?